packages/runtime/runtime-definitions/src/dataStoreFactory.ts

```diff
@@ -41,4 +41,13 @@ export interface IFluidDataStoreFactory extends IProvideFluidDataStoreFactory {
 		context: IFluidDataStoreContext,
 		existing: boolean,
 	): Promise<IFluidDataStoreChannel>;
+
+	/**
+	 * Synchronously creates a new runtime for a new data store from the data store context.
+	 *
+	 * @param context - Context for the data store.
+	 */
+	createDataStore?(context: IFluidDataStoreContext): {
```

Comment:
From a naming perspective, would probably prefer `instantiateDataStoreSync` to match `instantiateDataStore`.  `create` is a nicer verb if we were starting from scratch, but IMO it's more important to avoid implying some false distinction between the two to a prospective customer.

Reply:
instantiate is use for create and load, create is only used for create

Reply:
to expand a bit, this is only used to create an in memory detached datastore. if a handle to that datastore is stored in a dds, the datastore will attach, which will send an attach message, and the remote client will use the existing `instantiateDataStore` method to load it. Similarly, when the datastore is later loaded from the summary it will also use `instantiateDataStore`

Reply:
I see the distinction you're making now.

What implications does this have for using `instantiateDataStore` for creation then?  E.g. would we recommend folks to implement creation for both methods, recommend they favor the sync implementation and just call into it from `instantiateDataStore`, etc.?  Is it recommendable to build data stores that rely on async creation at all?  Just thinking about what we'd recommend as best practice if there are now two ways for the factory to create.

Reply:
I don't view it as an either/or. they are both useful. synchronous creation has a number of constraints around what can be done on the create path, not async, must be a child, must be synchronously available, and must support sync create. That said, it can also be necessary is some cases, which make those constraints acceptable. I would probably say the async path is preferred today, and sync should only be use in cases where it is necessary, and the consumer wants to deal with those constraints.

Reply:
I'll add some comments around this